plugin-localization/src/main/java/com.mapbox.mapboxsdk.plugins.localization/LocalizationPlugin.java

```diff
@@ -73,21 +79,36 @@ public String getDeviceLanguage() {
   }
 
   /**
-   * Sets map text to the specified language
+   * Checks whether a certain language is supported by OpenStreetMap.
    *
-   * @param languageToSetMapTo The language that to set the map text to.
-   * @since 0.1.0
+   * @param languageTag the language tag to check against OpenStreetMap-supported languages
+   * @return whether a certain language (tag) matches one of the tags associated with a
+   * OSM-supported language
+   */
+  private boolean openStreetMapSupportsLanguage(String languageTag) {
+    return languageTag.equals("ar")
+      || languageTag.equals("en")
+      || languageTag.equals("es")
+      || languageTag.equals("fr")
+      || languageTag.equals("de")
+      || languageTag.equals("pt")
+      || languageTag.equals("ru")
+      || languageTag.equals("zh")
+      || languageTag.equals("zh-Hans");
+  }
+
+  /**
+   * Retrieves and adjusts the individual map layers that need to be adjusted
+   *
+   * @param languageToSetMapTo the language that the map is being adjusted to
    */
-  public void setMapTextLanguage(String languageToSetMapTo) {
+  private void adjustMapTextLayers(String languageToSetMapTo) {
     for (Source source : mapboxMap.getSources()) {
       if (sourceIsFromMapbox(source)) {
         for (Layer layer : mapboxMap.getLayers()) {
           if (layerHasAdjustableTextField(layer)) {
-            if (((SymbolLayer) layer).getTextField().getValue().contains("name")) {
-              layer.setProperties(textField(String.format("{name_%s}", languageToSetMapTo)));
-            }
-            if (((SymbolLayer) layer).getTextField().getValue().contains("abbr")
-              && !getDeviceLanguage().equals("en")) {
+            if (((SymbolLayer) layer).getTextField().getValue().contains("name")
+              || ((SymbolLayer) layer).getTextField().getValue().equals("abbr")) {
```

Comment:
If the language is English and the text field is `{abbr}`, we should use `{abbr}`, because that field is written in English.